src/Modules/SimplCommerce.Module.Reviews/Areas/Reviews/Components/ReviewViewComponent.cs

```diff
@@ -57,7 +63,35 @@ public async Task<IViewComponentResult> InvokeAsync(long entityId, string entity
             model.Rating4Count = model.Items.Data.Count(x => x.Rating == 4);
             model.Rating5Count = model.Items.Data.Count(x => x.Rating == 5);
 
+            if (User.Identity.IsAuthenticated == true)
+            {
+                var user = await _workContext.GetCurrentUser();
+                model.LoggedUserName = user.FullName;
+
+                var currentUserOrder = _orderRepository
+                    .Query()
+                    .Where(x => x.CreatedById == user.Id);
+                var userOrderedProduct = currentUserOrder
+                    .Where(x => x.OrderItems.Where(y =>y.ProductId == entityId).Count() > 0);
+
+            
+   
+                 model.CanReview = userOrderedProduct.Any() ? true : false;
```

Comment:
I think CanReview is not a good name, it's too general. How about HasBoughtProduct?

Reply:
You don't need ternary operator, right?!

`model.HasBoughtProduct= userOrderedProduct.Any();`